doc/build_apps/crypto.rst

```diff
@@ -10,37 +10,87 @@ Hashing
 .. doxygenfunction:: crypto::SHA256
   :project: CCF
 
-.. doxygenclass:: crypto::HashProvider
+.. doxygenClass:: crypto::HashProvider
+  :project: CCF
+  :members:
+
+.. doxygenfunction:: crypto::make_hash_provider
   :project: CCF
 
 
-Asymmetric Cryptography
+Asymmetric Keys
 -----------------------
 
+CCF supports EC and RSA keys; public keys are held in (RSA)PublicKey objects and
+private keys in (RSA)KeyPair objects. (RSA)KeyPairs automatically generate random
+keys when constructed via :cpp:func:`KeyPairPtr crypto::make_key_pair(CurveID)` or
+:cpp:func:`RSAKeyPairPtr crypto::make_rsa_key_pair(size_t, size_t)`.
+
+.. doxygenclass:: crypto::PublicKey
+  :project: CCF
+  :members:
+
+.. doxygenclass:: crypto::KeyPair
+  :project: CCF
+  :members:
+
+.. doxygenclass:: crypto::RSAPublicKey
+  :project: CCF
+  :members:
+
+.. doxygenclass:: crypto::RSAKeyPair
+  :project: CCF
+  :members:
+
+.. doxygenfunction:: crypto::make_key_pair(CurveID)
```

Comment:
Can we add a .. doxygenenum:: crypto::CurveID?

Reply:
Sure!